### PR TITLE
add snap to CI

### PR DESCRIPTION
**Pull request by General-Beck**
initial build snap package for parity

### CODE REVIEW

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -60,6 +60,20 @@ linux-stable:
     - target/release/ethstore
     - target/release/ethkey
     name: "stable-x86_64-unknown-linux-gnu_parity"
+linux-snap:
+  stage: build
+  only:
+    - snap
+  script:
+    #- git clone https://github.com/paritytech/parity
+    - git submodule update --init --recursive
```
Comment (arkpar) [inline .gitlab-ci.yml]: This can be removed

File: .gitlab-ci.yml [hunk 2]
```diff
@@ -60,6 +60,23 @@ linux-stable:
...
+  image: ubuntu:latest
+  only:
+    - snap
+  before_script:
+    - apt install snapcraft
+  script:
+    - git submodule update --init --recursive
+    - cd scripts
+    - snapcraft
+    - sudo snap install *.snap --dangerous
```
Comment (rphmeier) [inline .gitlab-ci.yml]: what is the `--dangerous` flag used for?

Comment (arkpar) [inline .gitlab-ci.yml]: see https://github.com/paritytech/parity/pull/5496#issuecomment-296789185

Comment (General-Beck) [inline .gitlab-ci.yml]: @rphmeier We will not use this flag, it is only for installation

Comment (General-Beck) [inline .gitlab-ci.yml]: need for test snap package https://gitlab.ethcore.io/parity/parity/builds/53332/artifacts/download

File: .gitlab-ci.yml [hunk 3]
```diff
@@ -60,6 +60,23 @@ linux-stable:
     - target/release/ethstore
     - target/release/ethkey
     name: "stable-x86_64-unknown-linux-gnu_parity"
+linux-snap:
+  stage: build
+  image: ethcore/snapcraft:latest
```
Comment (rphmeier) [inline .gitlab-ci.yml]: ethcore should be migrated to paritytech at some point

Comment (General-Beck) [inline .gitlab-ci.yml]: @rphmeier I'm work on this https://hub.docker.com/r/parity

**Comment by elopio** [comment]
@General-Beck you are using gitlab? That's very nice!

I think you could configure this also to automate the deployment like we do in travis. You would have to encrypt the credentials [1], and then do something like this: https://github.com/elopio/ipfs-snap/blob/master/.travis.yml#L10

I haven't experimented with gitlab ci for this yet, so I'd love to help.

And there's also launchpad as an alternative, which we are currently migrating to a nicer UI in build.snapcraft.io: https://snapcraft.io/docs/build-snaps/ci-integration#using-launchpad

[1] https://docs.gitlab.com/ce/ci/variables/#secret-variables

**Comment by General-Beck** [comment]
@elopio Thanks, I'll try

**Pull request merged**